File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        string result = string(word1.length() + word2.length(), ' ');

        for (size_t i = 0, j = 0; i < word1.length() || i < word2.length(); ++i) {
            if (i < word1.length()) {
                result[j++] = word1[i];
            }
            if (i < word2.length()) {
                result[j++] = word2[i];
            }
        }

        return result;
    }
};

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.ts
function mergeAlternately(word1: string, word2: string): string {

    let i: number = 0;
    let merged: string = '';

    for (; i < word1.length && i < word2.length; ++i) {
        merged += word1[i] + word2[i];
    }

    merged += word1.substring(i) + word2.substring(i)
    
    return merged;
};